core/dictgen/src/rootcling_impl.cxx

```diff
@@ -3712,360 +3707,336 @@ static void MaybeSuppressWin32CrashDialogs() {
+static llvm::cl::list<std::string>
+OptIncludePaths("I", llvm::cl::Prefix, llvm::cl::ZeroOrMore,
+                llvm::cl::desc("Specify an include path."),
+                llvm::cl::cat(RootclingOptions));
+static llvm::cl::list<std::string>
+OptPPDefines("D", llvm::cl::Prefix, llvm::cl::ZeroOrMore,
+             llvm::cl::desc("Specify defined macros."),
+             llvm::cl::cat(RootclingOptions));
+static llvm::cl::list<std::string>
+OptWDiags("W", llvm::cl::Prefix, llvm::cl::ZeroOrMore,
+          llvm::cl::desc("Specify compiler diagnostics options."),
+          llvm::cl::cat(RootclingOptions));
+static llvm::cl::list<std::string>
+OptDictionaryHeaderFiles(llvm::cl::Positional, llvm::cl::OneOrMore,
+                         llvm::cl::desc("<list of dictionary header files> <LinkDef file>"),
+                         llvm::cl::cat(RootclingOptions));
+
+
 ////////////////////////////////////////////////////////////////////////////////
 
 int RootClingMain(int argc,
               char **argv,
               bool isGenreflex = false)
 {
+   // Define Options aliasses
+   auto &opts = llvm::cl::getRegisteredOptions();
+   auto &OptHelp = *opts["help"];
+   llvm::cl::alias OptHelpAlias1("h",
+                      llvm::cl::desc("Alias for -help"),
+                      llvm::cl::aliasopt(OptHelp));
+   llvm::cl::alias OptHelpAlias2("?",
+                      llvm::cl::desc("Alias for -help"),
+                      llvm::cl::aliasopt(OptHelp));
+
    // Copied from cling driver.
    // FIXME: Uncomment once we fix ROOT's teardown order.
    //llvm::llvm_shutdown_obj shutdownTrigger;
 
-   llvm::sys::PrintStackTraceOnErrorSignal(argv[0]);
+   const char *ExecutableFileName = argv[0];
+
+   llvm::sys::PrintStackTraceOnErrorSignal(ExecutableFileName);
    llvm::PrettyStackTraceProgram X(argc, argv);
    MaybeSuppressWin32CrashDialogs();
 
-   if (argc < 2) {
-      fprintf(stderr,
-              shortHelp,
-              argv[0]);
-      fprintf(stderr, "For more extensive help type: %s --help\n", argv[0]);
-      return 1;
+#if defined(R__WIN32) && !defined(R__WINGCC)
+   // FIXME: This is terrible hack allocating and changing the argument set.
+   // We should remove it and use standard llvm facilities to convert the paths.
+   // cygwin's make is presenting us some cygwin paths even though
+   // we are windows native. Convert them as good as we can.
+   for (int iic = 1 /* ignore binary file name in argv[0] */; iic < argc; ++iic) {
+      std::string iiarg(argv[iic]);
+      if (FromCygToNativePath(iiarg)) {
+         size_t len = iiarg.length();
+         // yes, we leak.
+         char *argviic = new char[len + 1];
+         strlcpy(argviic, iiarg.c_str(), len + 1);
+         argv[iic] = argviic;
+      }
    }
+#endif
+
+   // Hide options from llvm which we got from static initialization of libCling.
+   llvm::cl::HideUnrelatedOptions(/*keep*/RootclingOptions);
+
+   llvm::cl::ParseCommandLineOptions(argc, argv, "rootcling");
```

Comment:
Isn't `--lib-list-prefix` rejected because of `LongOptionsUseDoubleDash = false`? We still have long options starting with a single dash, e.g. `-gccxml` - how does the parser manage? Or do we have no test for `--lib-list-prefix`?

Reply:
All ACLiC tests use this option and we see no problems.

Reply:
I see no setting for single `-` vs double `--`. Are you saying `cl` eats both `--gccxml` and `-gccxml`, both `--lib-list-prefix` and `-lib-list-prefix`? Could you try? Is there a way to tell a `cl::option` to allow only `-` or only `--`?

Reply:
I can confirm that -gccxml and --gccxml both work. Simple search did not show a way to disallow `--` but I doubt it is not configurable. If we have no support we can easily implement it on our side.